VSDiagnostics/VSDiagnostics/VSDiagnostics/Utilities/Extensions.cs

```diff
@@ -205,5 +206,62 @@ public static bool IsNameofInvocation(this InvocationExpressionSyntax invocation
 
             return identifier != null && identifier.Identifier.ValueText == "nameof";
         }
+
+        public static List<T> SyntaxNodeOfType<T>(this IEnumerable<SyntaxNode> enumerable, SyntaxKind kind) where T : SyntaxNode
```

Comment:
Why not? The other overload doesn't have a `SyntaxKind` argument. Can you verify it?